code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1892,6 +1893,96 @@ GLOBAL_LIST_EMPTY(roundstart_races)
+	if((humi?.wear_suit?.clothing_flags & STOPSPRESSUREDAMAGE) && (humi?.head?.clothing_flags & STOPSPRESSUREDAMAGE))
+		return
+
+	for(var/gas_id in environment.gases)
+		var/gas_amount = environment.gases[gas_id][MOLES]
+		switch(gas_id)
+			if(/datum/gas/antinoblium)
+				if(gas_amount > 30)
+					humi.apply_status_effect(/datum/status_effect/corrosion_curse)
+			if(/datum/gas/freon)
+				if(isplasmaman(humi))
+					if(gas_amount > 5 && gas_amount < 10)
+						to_chat(humi, span_warning(pick("Your bones feels hot.", "You feel hot.")))
+					else if(gas_amount >= 10)
+						to_chat(humi, span_warning(pick("Your bones starts to heat up!", "You feel very hot!")))
+						humi.adjustStaminaLoss(- gas_amount * 0.15 * delta_time)
+						humi.adjust_bodytemperature(max(gas_amount * 0.25, BODYTEMP_HEATING_MAX) * delta_time)
+					continue
+				if(gas_amount > 5 && gas_amount < 10)
+					to_chat(humi, span_warning(pick("Your skin feels cold.", "You feel cold.")))
+				else if(gas_amount >= 10)
+					to_chat(humi, span_warning(pick("Your skin starts to freeze!", "You feel very cold!")))
+					var/current_stamina = humi.getStaminaLoss()
+					if(current_stamina > 25)
+						humi.adjustStaminaLoss(gas_amount * 0.15 * delta_time)
+					humi.adjust_bodytemperature(max(- gas_amount * 0.25, BODYTEMP_COOLING_MAX) * delta_time)
+					if(humi.bodytemperature < 265)
+						humi.apply_status_effect(/datum/status_effect/freon)
+			if(/datum/gas/halon)
+				if(gas_amount > 15)
+					humi.apply_status_effect(/datum/status_effect/stoned)
+			if(/datum/gas/healium)
+				if(gas_amount > 5)
+					to_chat(humi, span_notice("You feel better."))
+					humi.adjustBruteLoss(- gas_amount * 0.05 * delta_time)
+					humi.adjustFireLoss(- gas_amount * 0.015 * delta_time)
+					humi.adjustStaminaLoss(- gas_amount * 0.015 * delta_time)
+			if(/datum/gas/helium)
+				if(gas_amount > 15)
+					humi.apply_status_effect(/datum/status_effect/floaty)
+			if(/datum/gas/hypernoblium)
+				if(gas_amount > 5)
+					humi.apply_status_effect(/datum/status_effect/antimagic)
+			if(/datum/gas/nitryl)
+				if(ispodperson(humi))
+					humi.adjustToxLoss(- gas_amount * 0.5 * delta_time)
+					humi.adjustBruteLoss(- gas_amount * 0.5 * delta_time)
+					var/current_reagent = humi.reagents.get_reagent_amount(/datum/reagent/nitryl)
+					humi.reagents.add_reagent(/datum/reagent/nitryl, max(0, 1 - current_reagent))
+					continue
+				if(gas_amount > 5 && gas_amount < 10)
+					to_chat(humi, span_warning(pick("Your skin feels itchy.", "You feel like burning.")))
+				else if(gas_amount >= 10)
+					to_chat(humi, span_warning(pick("Your skin feels hot.", "You feel like burning.")))
+					humi.adjustFireLoss(gas_amount * 0.05 * delta_time)
+			if(/datum/gas/oxygen)
+				if(gas_amount > 50 && environment.temperature > 400)
+					humi.adjust_fire_stacks(gas_amount * 0.1)
+			if(/datum/gas/plasma)
```

Comment:
heals plasmamans (useful in a room without O2)

Reply:
Seems like a more specialized case of healium, plasmamen already like making smoking rooms and such, why not just leave it at that?

Reply:
i mean this would be very niche since plasmamans require to create the room to live in, i think they should be somehow rewarded for it and have an healing effect should be nice

Reply:
It doesn't make a lot of sense to me thinking about how plasmamen are always surrounded by plasma gas, but that's not too big a deal.

I don't really like tying healing to this stuff, since they can just sit in there forever, and that should be the property of med I think